### PR TITLE
(PC-8020) App Native - Amélioration tech - Ajouter le deviceId et le userId pour investigations

### PR DESCRIPTION
**Pull request by kopax-polyconseil**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-8020

## Checklist

I have:

- [x] Made sure the title of my PR follows the convention `[PC-XXX] <summary>`.
- [x] Made sure my feature is working on the relevant real / virtual devices.
- [ ] Written **unit tests** for my feature.
- [ ] Added new reusable components to the AppComponents page and communicate to the team on slack
- [ ] Added a **screenshot** for UI tickets.
- [ ] Attached a **ticket number** for any added TODO/FIXME \
       (for tech tasks, give the best context about the TODO resolution: what/who/when).

## Screenshots

| \*iOS - [Phone name] | \*Android - [Phone name] | Tablet - [Phone name] | Little - [Phone name] |
| -------------------- | :----------------------: | --------------------: | --------------------: |
|                      |                          |                       |                       |

## Deploy hard

If native code (ios/android) was modified, **after** the PR is merged, on the master branch, upgrade the **app version** (+1 patch):

- if you want an hard deployment of the testing environment, use `yarn trigger:testing:deploy`

### CODE REVIEW

**Review by BenStirrup — APPROVED** [review]

File: src/libs/errorMonitoring/ErrorMonitoringProvider.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    useEffect(() => {
+      // Last disconnected user id will be sent to Sentry on purpose until application restart
+      if (user) {
+        Sentry.configureScope((scope) => {
+          scope.setExtra('userId', user.id)
+        })
+      }
+    }, [user])
+
+    return <React.Fragment>{children}</React.Fragment>
```
Comment (antoinewg) [inline src/libs/errorMonitoring/ErrorMonitoringProvider.tsx:28]: le React.fragment est pas nécessaire

File: yarn.lock [hunk 2]
```diff
@@ -10176,6 +10176,11 @@ react-native-dash@^0.0.11:
     prop-types "^15.5.10"
     react-native-measureme "0.0.2"
 
+react-native-device-info@^8.1.0:
+  version "8.1.0"
+  resolved "https://npm.kopaxgroup.com:443/react-native-device-info/-/react-native-device-info-8.1.0.tgz#7d0ecf65cb26dbdbcdd555fc7bce2d9750b8b3d9"
```
Comment (antoinewg) [inline yarn.lock:10181]: resolved. pk pas `registry.yarnpkg` ?

File: android/app/src/main/java/com/passculture/MainApplication.java [hunk 3]
```diff
@@ -8,6 +8,9 @@
 import com.facebook.hermes.reactexecutor.HermesExecutorFactory;
 import com.facebook.react.bridge.JavaScriptExecutorFactory;
 import com.facebook.react.ReactApplication;
+import com.learnium.RNDeviceInfo.RNDeviceInfo;
+import com.learnium.RNDeviceInfo.RNDeviceInfo;
+import com.learnium.RNDeviceInfo.RNDeviceInfo;
```
Comment (antoinewg) [inline android/app/src/main/java/com/passculture/MainApplication.java:13]: 3 fois. ?

File: src/libs/errorMonitoring/ErrorMonitoringProvider.tsx [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+import React, { FC, memo, PropsWithChildren, useEffect } from 'react'
+
+import { useUserProfileInfo } from 'features/home/api'
+import { errorMonitoring } from 'libs/errorMonitoring/services'
+
+interface Props {
+  enabled?: boolean
+}
+
+export const ErrorMonitoringProvider: FC<PropsWithChildren<Props>> = memo(
```
Comment (antoinewg) [inline src/libs/errorMonitoring/ErrorMonitoringProvider.tsx:11]: tu peux expliquer l'utilisation du memo  ici ?

File: src/libs/errorMonitoring/ErrorMonitoringProvider.tsx [hunk 5]
```diff
@@ -0,0 +1,30 @@
+import * as Sentry from '@sentry/react-native'
+import React, { FC, memo, PropsWithChildren, useEffect } from 'react'
+
+import { useUserProfileInfo } from 'features/home/api'
+import { errorMonitoring } from 'libs/errorMonitoring/services'
+
+interface Props {
+  enabled?: boolean
```
Comment (antoinewg) [inline src/libs/errorMonitoring/ErrorMonitoringProvider.tsx:8]: pk pas ?
 
```suggestion
  enabled: boolean
```

File: src/libs/errorMonitoring/ErrorMonitoringProvider.tsx [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+    useEffect(() => {
+      errorMonitoring.init({ enabled: !!enabled })
+    }, [enabled])
+
+    useEffect(() => {
+      // Last disconnected user id will be sent to Sentry on purpose until application restart
+      if (user) {
+        Sentry.configureScope((scope) => {
+          scope.setExtra('userId', user.id)
```
Comment (antoinewg) [inline src/libs/errorMonitoring/ErrorMonitoringProvider.tsx:23]: il faut pas rajouter le device id dans sentry aussi ?

File: yarn.lock (hunk 2 above)
Comment (kopax-polyconseil) [reply yarn.lock:10181]: Erf c'est une erreur j'utilise un proxy npm privé

File: android/app/src/main/java/com/passculture/MainApplication.java (hunk 3 above)
Comment (kopax-polyconseil) [reply android/app/src/main/java/com/passculture/MainApplication.java:13]: yes erreur

File: src/libs/errorMonitoring/ErrorMonitoringProvider.tsx (hunk 4 above)
Comment (kopax-polyconseil) [reply src/libs/errorMonitoring/ErrorMonitoringProvider.tsx:11]: Cela évite de rerender le composant si les props change.

File: src/libs/errorMonitoring/ErrorMonitoringProvider.tsx (hunk 6 above)
Comment (kopax-polyconseil) [reply src/libs/errorMonitoring/ErrorMonitoringProvider.tsx:23]: C'est déjà envoyé automatiquement par Sentry

File: src/libs/errorMonitoring/ErrorMonitoringProvider.tsx (hunk 5 above)
Comment (kopax-polyconseil) [reply src/libs/errorMonitoring/ErrorMonitoringProvider.tsx:8]: OK

**Pull request merged**